src/data/load_geometry.js

```diff
@@ -48,40 +47,57 @@ function resample(ring) {
  */
 export default function loadGeometry(feature: VectorTileFeature, canonical): Array<Array<Point>> {
     if (!canonical) return [];
```

Comment:
In my PR, I had to delete this to get unit tests to pass. Instead of returning an empty array, we need to retain the original logic if `canonical` is undefined/null and avoid reprojecting unnecessarily. We can refactor `reproject` to do this without changing the rest of the logic in `loadGeometry`.

```
function reproject(p, featureExtent, canonical) {
    let x, y;
    if (canonical) {
        const cs = projection.tileTransform(canonical);
        const z2 = Math.pow(2, canonical.z);
        const lng = lngFromMercatorX((canonical.x + p.x / featureExtent) / z2);
        const lat = latFromMercatorY((canonical.y + p.y / featureExtent) / z2);
        const projectedLngLat = projection.project(lng, lat);
        x = (projectedLngLat.x * cs.scale - cs.x) * EXTENT;
        y = (projectedLngLat.y * cs.scale - cs.y) * EXTENT;
    } else {
        const scale = EXTENT / featureExtent;
        x = Math.round(p.x * scale);
        y = Math.round(p.y * scale);
    }

    return new Point(x, y);
}
```